test/blackbox-tests/test-cases/pkg/vendored-vs-data-dir.t

```diff
@@ -0,0 +1,96 @@
+Showcase the difference in Dune package management solving dependencies of
+vendored directories and data directories. In both cases, the top level project
+doesn't use the vendored packages as a dependency.
+
+We create the necessary packages first
+
+  $ mkrepo
+
+  $ mkdir vendor
+  $ mkdir vendor/a
+  $ mkdir vendor/b
+
+  $ mkpkg lwt
+  $ mkpkg eio
+  $ mk_ocaml 5.2.0
```

Comment:
Do we need the `ocaml` dependency here? I think it should be possible to demonstrate the same without the fake compilers.